File: exttool/ControllerInput.cs
using System;
using System.Collections.Generic;

using BizHawk.Client.Common;

namespace BirdsEye {
    public class ControllerInput {
        private readonly bool[] _inputState = {false, false, false, false, false, false, false, false, false, false, false, false};
        private readonly Logging _log;

        public ControllerInput(Logging log) {
            _log = log;
        }

        ///<summary>
        /// Execute the current input state in the emulator.
        ///</summary>

        public void ExecuteInput(ApiContainer APIs) {
            APIs.Joypad.Set(new Dictionary<string, bool>() {
                {"A", _inputState[0]},
                {"B", _inputState[1]},
                {"X", _inputState[2]},
                {"Y", _inputState[3]},
                {"L", _inputState[4]},
                {"R", _inputState[5]},
                {"Up", _inputState[6]},
                {"Down", _inputState[7]},
                {"Right", _inputState[8]},
                {"Left", _inputState[9]},
                {"Start", _inputState[10]},
                {"Select", _inputState[11]}
            }, 1);
        }

        ///<summary>
        /// Set the controller input from a string containing the state of different buttons
        /// to be executed when `ExecuteInput` is called.
        ///</summary>
        public void SetInputFromString(string str) {
            string[] newState = str.Trim(';').Split(';');
            for (int i = 0; i < 12; i++) {
                _inputState[i] = Convert.ToBoolean(newState[i]);
            }
        }
    }
}

File: birdseyelib/controllerInput.py
class ControllerInput:
    """Class containing various functions for reading memory from BizHawk."""
    def __init__(self, client) -> None:
        self.client = client

    def set_controller_input(self, a=False, b=False, x=False, y=False, l=False, r=False, up=False, down=False, right=False, left=False, start=False, select=False):
        """Sets the controller inputs to be executed in the emulator.
        All inputs are set to `False` be default.
        The inputs are executed until a new controller input is sent.

        :param a: The state of the A button.
        :type a: bool

        :param b: The state of the B button.
        :type b: bool

        :param x: The state of the X button on the control pad.
        :type x: bool
        
        :param y: The state of the Y button on the control pad.
        :type y: bool

        :param l: The state of the L shoulder button on the control pad.
        :type l: bool

        :param r: The state of the R shoulder button on the control pad.
        :type r: bool

        :param up: The state of the Up button on the control pad.
        :type up: bool

        :param down: The state of the Down button on the control pad.
        :type down: bool

        :param right: The state of the Right button on the control pad.
        :type right: bool

        :param left: The state of the Left button on the control pad.
        :type left: bool

        :param start: The state of the Start button on the control pad.
        :type start: bool
        
        :param select: The state of the Select button on the control pad.
        :type select: bool"""
        bool_to_string = {False : "false", True : "true"}
        controller_input = bool_to_string[a] + ";" + bool_to_string[b] + ";" + \
                           bool_to_string[x] + ";" + bool_to_string[y] + ";" + \
                           bool_to_string[l] + ";" + bool_to_string[r] + ";" + \
                           bool_to_string[up] + ";" + bool_to_string[down] + ";" + \
                           bool_to_string[right] + ";" + bool_to_string[left] + ";" +\
                           bool_to_string[start] + ";" + bool_to_string[select]
        self.client._queue_request("INPUT;" + controller_input + "\n")
